src/async_wrap.cc

```diff
@@ -430,6 +429,7 @@ void AsyncWrap::GetProviderType(const FunctionCallbackInfo<Value>& args) {
 
 
 void AsyncWrap::EmitDestroy() {
+  resource_.Reset();
   AsyncWrap::EmitDestroy(env(), async_id_);
```

Comment:
What if the async resource is garbage collected between entering `EmitDestroy()` and when the userland `destroy` hook runs? I think these two lines should be switched, right?